src/nni_manager/training_service/remote_machine/shellExecutor.ts

```diff
@@ -73,43 +108,120 @@ class ShellExecutor {
         this.sshClient.end();
     }
 
+    public addUsage(): boolean {
+        let isAddedSuccess = false;
+        if (this.usedCount < this.maxUsageCount) {
+            this.usedCount++;
+            isAddedSuccess = true;
+        }
+        return isAddedSuccess;
+    }
+
+    public releaseUsage(): boolean {
+        let canBeReleased = false;
+        if (this.usedCount > 0) {
+            this.usedCount--;
+        }
+        if (this.usedCount == 0) {
+            canBeReleased = true;
+        }
+        return canBeReleased;
+    }
+
     public get getUsedConnectionNumber(): number {
-        return this.usedConnectionNumber;
+        return this.usedCount;
     }
 
     public addUsedConnectionNumber(): void {
```

Comment:
should this function `addUsedConnectionNumber`  be removed?

Reply:
fixed